packages/wrangler/src/autoconfig/frameworks/tanstack.ts

```diff
@@ -0,0 +1,143 @@
+import assert from "assert";
+import path from "path";
+import { brandColor, dim } from "@cloudflare/cli/colors";
+import * as recast from "recast";
+import { transformFile } from "../c3-vendor/codemod";
+import { installPackages } from "../c3-vendor/packages";
+import { usesTypescript } from "../uses-typescript";
+import { Framework } from ".";
+import type { ConfigurationOptions } from ".";
+import type { RawConfig } from "@cloudflare/workers-utils";
+import type { types } from "recast";
+
+const b = recast.types.builders;
+const t = recast.types.namedTypes;
+
+function isPluginsProp(
+	prop: unknown
+): prop is types.namedTypes.ObjectProperty | types.namedTypes.Property {
+	return (
+		(t.Property.check(prop) || t.ObjectProperty.check(prop)) &&
+		t.Identifier.check(prop.key) &&
+		prop.key.name === "plugins"
+	);
+}
+
+function transformViteConfig(projectPath: string) {
+	const filePath = path.join(
+		projectPath,
+		`vite.config.${usesTypescript(projectPath) ? "ts" : "js"}`
```

Comment:
Just a note, possibly something we might want consider down the line, a TS project could also have a vite.config.js file, maybe we might want to check all types? 🤔 

It's just a thought, I think this might be pretty edge-case anyways 🙂

Reply:
I'm not sure that is an edge case. I'd imagine it's quite common for TypeScript projects to still have `vite.config.js` files. I think it needs to check both.

Reply:
I've made it check both in https://github.com/cloudflare/workers-sdk/pull/11375/commits/da7ee5ee7d999f7fa596c1bd1a3ee17743a3f8a1